packages/flutter_tools/lib/src/commands/build_ios.dart

```diff
@@ -104,13 +105,17 @@ class BuildIOSArchiveCommand extends _BuildIOSSubCommand {
     argParser.addOption(
       'export-method',
       defaultsTo: 'app-store',
-      allowed: <String>['app-store', 'ad-hoc', 'development', 'enterprise'],
+      allowed: <String>['app-store', 'app-store-connect', 'ad-hoc', 'release-testing', 'development', 'debugging', 'enterprise'],
```

Comment:
This pile is really confusing.  Can we keep the old allowed list as-is, but pass the right thing into Xcode based on the version?

Reply:
makes sense, done!